layouts/joomla/edit/frontediting_modules.php

```diff
@@ -25,52 +24,49 @@
 	$target  = '_self';
 }
 
-// Add class, editing URL and tooltip, and if module of type menu, also the tooltip for editing the menu item:
+// Add link for editing the module
 $count = 0;
 $moduleHtml = preg_replace(
-	// Replace first tag of module with a class
-	'/^(\s*<(?:div|span|nav|ul|ol|h\d|section|aside|nav|address|article) [^>]*class="[^"]*)"/',
-	// By itself, adding class jmoddiv and data attributes for the URL and tooltip:
-	'\\1 jmoddiv" data-jmodediturl="' . $editUrl . '" data-target="' . $target . '" data-jmodtip="'
-	.	HTMLHelper::_('tooltipText',
-			Text::_('JLIB_HTML_EDIT_MODULE'),
-			htmlspecialchars($mod->title, ENT_COMPAT, 'UTF-8') . '<br>' . sprintf(Text::_('JLIB_HTML_EDIT_MODULE_IN_POSITION'), htmlspecialchars($position, ENT_COMPAT, 'UTF-8')),
-			0
-		)
-	. '"'
-	// And if menu editing is enabled and allowed and it's a menu module, add data attributes for menu editing:
-	.	($menusEditing && $mod->module === 'mod_menu' ?
-			' data-jmenuedittip="' . HTMLHelper::_('tooltipText', 'JLIB_HTML_EDIT_MENU_ITEM', 'JLIB_HTML_EDIT_MENU_ITEM_ID') . '"'
-			:
-			''
-		),
+	// Find first tag of module
+	'/^(\s*<(?:div|span|nav|ul|ol|h\d|section|aside|address|article|form) [^>]*>)/',
+	// Create and add the edit link and tooltip
+	'\\1 <a class="btn btn-link jmodedit" href="' . $editUrl . '" target="' . $target . '" aria-describedby="tip-' . (int) $mod->id . '">
+	<span class="icon-edit" aria-hidden="true"></span>' . Text::_('JGLOBAL_EDIT') . '</a>
+	<div role="tooltip" id="tip-' . (int) $mod->id . '">' . Text::_('JLIB_HTML_EDIT_MODULE') . '<br>' . htmlspecialchars($mod->title, ENT_COMPAT, 'UTF-8') . '<br>'. sprintf(Text::_('JLIB_HTML_EDIT_MODULE_IN_POSITION'), htmlspecialchars($position, ENT_COMPAT, 'UTF-8')) . '</div>',
 	$moduleHtml,
 	1,
 	$count
 );
 
-if ($count)
+// If menu editing is enabled and allowed and it's a menu module add link for editing
+if ($menusEditing && $mod->module === 'mod_menu')
 {
-	// Load stylesheet and javascript to head:
-	HTMLHelper::_('bootstrap.popover');
+	// find the menu item id
+	$regex = '/\bitem-(\d+)\b/';
 
-	/** @var Joomla\CMS\WebAsset\WebAssetManager $wa */
-	$wa = Factory::getApplication()->getDocument()->getWebAssetManager();
-	$wa->usePreset('joomla.frontediting');
+	preg_match_all($regex, $moduleHtml, $menuItemids);
+	if ($menuItemids)
+	{
+		foreach ($menuItemids[1] as $menuItemid)
+			{
+				$menuitemEditUrl = Uri::base() . 'administrator/index.php?option=com_menus&view=item&client_id=0&layout=edit&id=' . (int) $menuItemid;
+				$moduleHtml = preg_replace(
+					// Find the link
+					'/(<li.*?\bitem-'. $menuItemid .'.*?>)/',
+					// Create and add the edit link
+					'\\1 <a class="small" href="' . $menuitemEditUrl . '" target="' . $target . '" title="' . Text::_('JLIB_HTML_EDIT_MENU_ITEM') . ' ' . sprintf(Text::_('JLIB_HTML_EDIT_MENU_ITEM_ID'), (int) $menuItemid) . '">
```

Comment:
```suggestion
					'\\1 <a class="jmenuitemedit small" href="' . $menuitemEditUrl . '" target="' . $target . '" title="' . Text::_('JLIB_HTML_EDIT_MENU_ITEM') . ' ' . sprintf(Text::_('JLIB_HTML_EDIT_MENU_ITEM_ID'), (int) $menuItemid) . '">
```
If we add a specific CSS class to that edit link, we can do something like the following in the SCSS file:

> a.jmenuitemedit {
>   margin-right: .5em;
> 
>   [dir="rtl"] & {
>     margin-right: 0;
>     margin-left: .5em;
>   }
> }

Reply:
Not sure if to use em or rem. I've tried with .5em and it looks better than now.

Reply:
Anyway a specific class would make styling easier.